python_modules/dagster/dagster/_config/pythonic_config/__init__.py

```diff
@@ -1795,7 +1799,7 @@ def separate_resource_params(cls: Type[BaseModel], data: Dict[str, Any]) -> Sepa
     """
     keys_by_alias = {field.alias: field for field in cls.__fields__.values()}
     data_with_annotation: List[Tuple[str, Any, Type]] = [
-        (k, v, keys_by_alias[k].annotation) for k, v in data.items() if k in keys_by_alias
+        (k, v, keys_by_alias[k].annotation if k in keys_by_alias else None) for k, v in data.items()
```

Comment:
is this related to the extra params fix?

Reply:
Yeah - this would actually filter out the extra params (because of the if clause) before it was being passed to Pydantic. This would mean the user-facing error wouldn't show up (it would just silently remove the extra params).

Reply:
ok cool - i just couldn't figure out where it fit in with the rest of the changes, but that makes sense!